test/e2e/operator_test.go

```diff
@@ -348,13 +348,20 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
 				t.Logf("Waiting for dns record: %s", expectedHost)
 				return false, nil
 			}
-			t.Logf("Got IPs: %v", ips)
+			gotIPs := make(map[string]struct{})
+			for _, ip := range ips {
+				gotIPs[ip] = struct{}{}
+			}
+			t.Logf("Got IPs: %v", gotIPs)
+
 			// If all IPs of the loadbalancer are not present query again.
-			if len(ips) < len(serviceIPs) {
+			if len(gotIPs) < len(serviceIPs) {
 				return false, nil
 			}
-			for _, ip := range ips {
-				if _, ok := serviceIPs[ip]; !ok {
+			// all expected IPs should be in the received IPs
+			// but these 2 sets are not necessary equal
+			for ip := range serviceIPs {
+				if _, found := gotIPs[ip]; !found {
 					return false, nil
 				}
 			}
```

Comment:
@arjunrn: this is to fix [this issue](https://github.com/openshift/external-dns-operator/pull/104#issuecomment-995174939). It doesn't happen all the time but when it does, the test times out.

Reply:
I would like to see this change in a separate PR if possible. Also `gotIPs` could be called `loadbalancerIPs`.